app/models/signup.rb

```diff
@@ -17,15 +18,20 @@ class Signup < ApplicationRecord
   validates :phone_number, phone: true, allow_blank: true, format: { with: /\A\+1[0-9]{10}\z/ }
   validates :email_address, 'valid_email_2/email': true
 
-  def self.valid_emails
-    distinct(:email_address).pluck(:email_address).filter do |email|
-      ValidEmail2::Address.new(email).valid?
+  def self.valid_emails_with_unsent_followups
+    distinct(:email_address).where(sent_followup: false).pluck(:email_address, :name).filter do |signup_info|
+      ValidEmail2::Address.new(signup_info[0]).valid?
     end
   end
 
-  def self.send_followup_emails
-    valid_emails.each_with_index do |email, index|
-      SignupFollowupMailer.followup(email).deliver_later(wait: index * 2)
+  def self.send_followup_emails(batch_size = nil)
+    emails_to_send = batch_size.present? ? valid_emails_with_unsent_followups.slice(0, batch_size) : valid_emails_with_unsent_followups
+
+    emails_to_send.each_with_index do |signup_info, index|
+      SignupFollowupMailer.followup(signup_info[0], signup_info[1]).deliver_later(wait: index * 2)
```

Comment:
deliver_later should enqueue these as jobs, so i dont _think_ we need to explicitly tell it to wait longer to process. But I also didn't know you could pass a "wait" and I'm interested to learn from you what it does!

Reply:
I worked on this a little - the idea is to spread these out over time so we don't overload the queue with these low-priority tasks.